apps/filebrowser/src/filebrowser/tasks.py

```diff
@@ -121,63 +136,108 @@ def document_cleanup_task(**kwargs):
     kwargs["state"] = "SUCCESS"
     LOG.info(f"Document_cleanup_task completed successfully.")
   except Exception as err:
+    kwargs["state"] = "FAILURE"
     document_cleanup_task.update_state(task_id=task_id, state='FAILURE', meta=kwargs)
     raise Exception(f"Upload failed %s" % err)
 
   kwargs["state"] = "SUCCESS"
   kwargs["progress"] = "100%"
-  kwargs["task_end"] = now.strftime("%Y-%m-%dT%H:%M:%S")
+  kwargs["task_end"] = timezone.now().astimezone(pytz.timezone(TIME_ZONE)).strftime("%Y-%m-%dT%H:%M:%S.%f")
   document_cleanup_task.update_state(task_id=task_id, state='SUCCESS', meta=kwargs)
 
   return None
 
+
 @app.task()
 def check_disk_usage_and_clean_task(**kwargs):
   task_id = kwargs.get("qquuid")
-  default_cleanup_threshold = 90
-  cleanup_threshold = kwargs.get('cleanup_threshold', default_cleanup_threshold)
+  cleanup_threshold = kwargs.get('cleanup_threshold', TASK_SERVER_V2.DISK_USAGE_CLEANUP_THRESHOLD)
   username = kwargs.get("username", "celery_scheduler")
-  now = timezone.now()
+  now = timezone.now().astimezone(pytz.timezone(TIME_ZONE))
   kwargs = {
     "username": username,
     "task_name": "tmp_cleanup",
     "task_id": task_id,
     "parameters": cleanup_threshold,
     "progress": "0%",
-    "task_start": now.strftime("%Y-%m-%dT%H:%M:%S")
+    "task_start": now.strftime("%Y-%m-%dT%H:%M:%S.%f"),
   }
 
+  check_disk_usage_and_clean_task.update_state(task_id=task_id, state=states.STARTED, meta=kwargs)
+
+  def delete_old_files(directory, current_time, time_delta):
+    for root, dirs, files in os.walk(directory):
+      for file in files:
+        file_path = os.path.join(root, file)
+        try:
+          file_modified_time = datetime.fromtimestamp(os.path.getmtime(file_path))
+          file_modified_time = timezone.make_aware(file_modified_time, timezone.get_default_timezone())
+          if file_modified_time < current_time - time_delta:
+            if os.path.isfile(file_path) or os.path.islink(file_path):
+              os.unlink(file_path)
+              LOG.info(f"Deleted file {file_path}")
+        except PermissionError:
+          LOG.warning(f"Permission denied: unable to delete {file_path}")
```

Comment:
What do you do with "Permission denied" error?

Reply:
We just log the permission denied error.